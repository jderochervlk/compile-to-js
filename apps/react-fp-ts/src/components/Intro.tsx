const Intro = () => (
  <>
    <h1 className="text-4xl">React and fp-ts</h1>
    <p className="mt-2">
      A basic application written with React, TypeScript, and{' '}
      <a
        href="https://gcanti.github.io/fp-ts/"
        className="text-blue-500 hover:text-blue-800 hover:underline"
      >
        fp-ts
      </a>
      .
    </p>
    <p>
      TypeScript with fp-ts is my standard way of writing TypeScript and I can't
      imagine not having Option and Either. I can avoid using null and
      undefined, and being able to pipe data is very nice. fp-ts feels like an
      extension and a standard library for TypeScript and I will never again
      write TypeScript without it.
    </p>
    <br />
  </>
)

export default Intro
