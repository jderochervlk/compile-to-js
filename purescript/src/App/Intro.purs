module App.Intro where

import Prelude
import Halogen as H
import Halogen.HTML as HH
import Halogen.HTML.Properties as HP

type State
  = Unit

initialState :: forall input. input -> State
initialState _ = unit

component :: forall query input output m. H.Component query input output m
component =
  H.mkComponent
    { initialState
    , render
    , eval: H.mkEval H.defaultEval
    }

header :: forall w i. HH.HTML w i
header =
  HH.div
    [ HP.classes
        [ HH.ClassName "container"
        , HH.ClassName "mx-auto"
        , HH.ClassName "p-4"
        , HH.ClassName "bg-white"
        , HH.ClassName "h-screen"
        ]
    ]
    [ HH.h1
        [ HP.classes [ HH.ClassName "text-4xl" ] ]
        [ HH.text $ "PureScript and Halogen?!?!?!?!" ]
    , HH.p
        [ HP.classes [ HH.ClassName "mt-2" ] ]
        [ HH.text $ " A basic application written with PureScript and Halogen." ]
    ]

render :: forall w i. State -> HH.HTML w i
render __ = header
