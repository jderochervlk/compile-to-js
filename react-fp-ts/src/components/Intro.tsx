const Intro = () => (
  <>
    <h1 className="text-4xl">React and fp-ts</h1>
    <p className="mt-2">
      A basic application written with React, TypeScript, and fp-ts.
    </p>
  </>
)

export default Intro
