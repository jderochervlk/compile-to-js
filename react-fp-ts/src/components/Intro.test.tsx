import { render } from '@testing-library/react'
import Intro from './Intro'

test('Intro contains the expected text', () => {
  const { container, getByText } = render(<Intro />)
  expect(container).toMatchSnapshot()
  expect(getByText('React and fp-ts')).toBeTruthy()
  expect(
    getByText('A basic application written with React, TypeScript, and fp-ts.')
  ).toBeTruthy()
})
