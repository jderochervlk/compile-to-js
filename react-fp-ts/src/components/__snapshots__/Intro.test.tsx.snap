// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro contains the expected text 1`] = `
<div>
  <h1
    class="text-4xl"
  >
    React and fp-ts
  </h1>
  <p
    class="mt-2"
  >
    A basic application written with React, TypeScript, and fp-ts.
  </p>
  <br />
</div>
`;
